components/rtos/rtos_memory_task.c

```diff
@@ -0,0 +1,67 @@
+// Copyright 2022 PWrInSpace, Kuba
+#include <string.h>
+#include <memory.h>
+#include "sdcard.h"
+#include "rtos_tasks.h"
+
+#define TAG "MEM_TASK"
+#define FILE_NAME "/sdcard/data"
+extern sd_card_t sd_card;
+
+static bool create_path_to_file(char *file_path, size_t size) {
+    char *path = (char*)calloc(size, sizeof(char));
+
+    for (int i = 0; i < 100; ++i) {
+        snprintf(path, size, "%s%d.txt", file_path, i);
+        if (SD_file_exists(path) == false) {
+            memcpy(file_path, path, size);
+            free(path);
+            return true;
+        }
+    }
+
+    free(path);
+    return false;
+}
+
+static bool can_save_data_to_sd(DATA_SAVE_OPTIONS option) {
+    if (option == SAVE_TO_BOTH || option == SAVE_TO_SD) {
+        return true;
+    }
+
+    return false;
+}
+
+static bool can_save_data_to_flash(DATA_SAVE_OPTIONS option) {
+    if (option == SAVE_TO_BOTH || option == SAVE_TO_FLASH) {
+        return true;
+    }
+
+    return false;
+}
+
+void memory_task(void *arg) {
+    char data_string[512];
+    data_to_memory_task_t data_to_save;
+    char file_path[sizeof(FILE_NAME) + 6] = FILE_NAME;
```

Comment:
sizeof(FILE_NAME) wywala rozmiar char* czyli 8, rozumiem ze stad te +6

Reply:
nie daje rozmiar całego FILE NAME, ale trzeba jeszcze do niego numer dopisać + .txt, więc w sumie te 6 to mało

Reply:
Nie no mordo będę się kłócił xD
sizeof to nie strlen, nawet sobie odpalałem kompilator obok i sprawdzałem

Reply:
dobra cofam
xDDDDD

Reply:
przy strlen musisz mieć ciąg znaków zakończony \0